app/controllers/votes_controller.rb

```diff
@@ -12,26 +12,8 @@ def create
 
     respond_to do |format|
       format.js {
-      if existing_vote.size > 0
-        if existing_vote.first.upvote == vote.upvote
-          existing_vote.first.destroy
-        else
-          existing_vote.first.destroy
-          if vote.save
-            @success = true
-          else
-            @success = false
-          end
-        end
-      else
-        if vote.save
-          @success = true
-        else
-          @success = false
-        end
-      end
-
-
+        existing_vote.first.destroy if existing_vote.size > 0
+        vote.save unless existing_vote.size > 0 && existing_vote.first.upvote == vote.upvote
```

Comment:
why do you have these logics under format.js ?

Reply:
i tried to do these logics in model magesh but its not working magesh.It is working when we have it here only magesh

Reply:
> i tried to do these logics in model magesh but its not working magesh.It is working when we have it here only magesh

why not? it should work. This one right here is not a good practice, so figure it out.

Reply:
This issue is being worked on by aboorva